content/docs/specs/native_histograms.md

```diff
@@ -0,0 +1,2353 @@
+humans to easily spot characteristics of distributions as they change over
+time.
+
+TODO: Heatmaps are not implemented yet. Instead, the UI plots just the sum of
+observations as a conventional graph. See [tracking
+issue](https://github.com/prometheus/prometheus/issues/11268). The same issue
+also discusses how to deal with the rendering of range vectors in the _Table_
+view.
+
+## Template expansion
+
+Native histograms work in template expansion. They are rendered in a text
+representation inspired by the mathematical notation of open and closed
+intervals. (This is generated by the `FloatHistogram.String` method in Go.) As
+native histograms can have a lot of buckets and bucket boundaries tend to have
+boundaries with a lot of decimal places, the representation isn't necessarily
+very readable. Use native histograms in template expansion judiciously.
+
+Example for the text representation of a float histogram:
+
+```
+{count:3493.3, sum:2.349209324e+06, [-22.62741699796952,-16):1000, [-16,-11.31370849898476):123400, [-4,-2.82842712474619):3, [-2.82842712474619,-2):3.1, [-0.01,0.01]:5.5, (0.35355339059327373,0.5]:1, (1,1.414213562373095]:3.3, (1.414213562373095,2]:4.2, (2,2.82842712474619]:0.1}
+```
+
+## Remote write & read
+
+The [protobuf specs for remote write &
+read](https://github.com/prometheus/prometheus/blob/main/prompb) were extended
+for native histograms as an experimental feature. Receivers not capable of
+processing native histograms will simply ignore the newly added fields.
+Nevertheless, Prometheus has to be configured to send native histograms via
+remote write (by setting the `send_native_histograms` remote write config
+setting to true).
+
+In [remote write v2](https://prometheus.io/docs/specs/remote_write_spec_2_0/),
+native histograms are a stable feature.
+
+TODO: Is there more to say? I'm not a remote write & read specialist. Does
+remote read really work?
+
+TODO: What are the plans to auto-convert classic histograms to NHCBs in
+remote-write?
+
+## Federation
+
+Federation of native histograms works as expected, provided the federation
+scrape uses the protobuf format. A federation via OpenMetrics text format will
+be possible, at least in principle, once native histograms are supported in
+that format, but federation via protobuf is preferred for efficiency reasons
+anyway.
+
+TODO: Clarify state of federation of NHCBs. Update once OM supports NH.
+
+## OTLP
+
+The OTLP receiver built into Prometheus will convert incoming OTel exponential
+histograms to Prometheus native histograms utilizing the compatibility
+described [above](#opentelemetry-interoperability). The resolution of a
+histogram using a schema (“scale” in OTel lingo) greater than 8 will be reduced
+to match schema 8. (In the unlikely case that a schema smaller than -4 is used,
+the ingestion will fail.)
+
+TODO: Is the OTLP receiver documented anywhere? Link the documentation here.
```

Comment:
+ TODO convert to NHCB explicit bucket histograms. https://github.com/prometheus/prometheus/issues/15022

Reply:
@aknuds1  about documentation